renderdoc/os/posix/posix_stringio.cpp

```diff
@@ -141,10 +141,14 @@ void GetDefaultFiles(const char *logBaseName, string &capture_filename, string &
   GetExecutableFilename(path);
 
   const char *mod = strrchr(path.c_str(), '/');
-  if(mod == NULL)
-    mod = "unknown";
-  else
+  if(mod != NULL)
     mod++;
+  else
+#if defined(RENDERDOC_PLATFORM_ANDROID)
+    mod = path.c_str();    // Keep as org.company.app
+#else
+    mod = "unknown";
+#endif
```

Comment:
The whole point of having OS-specific code split out into files is to avoid `#ifdef` cruft from building up.

In this case the only reason for the "unknown" before was that the filename was expected to actually be a filename, so it would always include a `/`. It's perfectly reasonable just to change it so that on all platforms it first tries to extract the filename from a path, and then if it doesn't find a `/` to fall back to using the filename as-is, with a final check to see if the path is empty at which point it falls back to "unknown". That way there's no need for `#ifdef`s here and we still always get some kind of valid filename.